lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2185,4 +2227,14 @@ export class UserAgentApplication {
         };
         return new TelemetryManager(telemetryManagerConfig, telemetryEmitter);
     }
+
+    /**
+     * Helper function to clear the cache and throw an error
+     * @param storage
+     * @param error
+     */
+    private returnErrorAndClearTempCache(storage: AuthCache, error: AuthError, state: string) {
```

Comment:
this method feels confusing to me, I understand what it is doing but it feels like there must be a cleaner way. 

Really `storage.resetTempCacheItems(state);` is all you are doing.
I would almost say just put that line in every catch instead of this method that takes errors and storage as well.

```js
.catch(err => {
       this.storage.resetTempCacheItems(state)
       throw err; 
 });
```

I do like that you are not coping the same code and can update it in one place, it just feels like it mixes up responsiblities.

Reply:
Yes. The method initially was conceived as we are handling it at each throw. Now that we are bubbling up the errors in all use cases, guess we can do this.

Reply:
Changes done.